### PR TITLE
[site] Added pprof doc

### LINKED ISSUE #983: Include the documentation for using pprof
Add instructions on how to connect to pprof
ref: https://github.com/kubernetes-sigs/kueue/pull/978#discussion_r1262632303

/kind documentation

### PR DESCRIPTION
**Pull request by stuton**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
[site] Added pprof doc

#### Which issue(s) this PR fixes:

Fixes #983

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: site/content/en/docs/installation/_index.md [hunk 1]
```diff
@@ -206,3 +207,21 @@ To uninstall Kueue, run the following command:
...
+
+```shell
+kubectl port-forward kueue-controller-manager-769f96b5dc-87sf2 -n kueue-system 8082:8082
+Forwarding from 127.0.0.1:8082 -> 8082
+Forwarding from [::1]:8082 -> 8082
+```
+
+The HTTP endpoint will now be available as a local port.
+
+Please see [basic usage](https://github.com/google/pprof#basic-usage) pprof command and [examples](https://pkg.go.dev/net/http/pprof#hdr-Usage_examples) of using endpoints
```
Comment (trasc) [inline site/content/en/docs/installation/_index.md:227]:
```suggestion
Please see [basic usage](https://github.com/google/pprof#basic-usage) pprof command and [examples](https://pkg.go.dev/net/http/pprof#hdr-Usage_examples) of using the exposed endpoints.
```

**Review by trasc — COMMENTED** [review]
nit, otherwise LGTM

**Comment by alculquicondor** [comment]
@moficodes the bot is not adding a preview for this PR. What do we need to do?

File: site/content/en/docs/installation/_index.md [hunk 2]
```diff
@@ -206,3 +207,21 @@ To uninstall Kueue, run the following command:
 ```sh
 make undeploy
 ```
+
+## Enabling pprof endpoints
+
+> _Available in Kueue v0.5.0 and later_
+
+It's available only if `pprofBindAddress` is set in the [manager's configuration](/docs/installation/#install-a-custom-configured-released-version).
```
Comment (alculquicondor) [inline site/content/en/docs/installation/_index.md:215]:
```suggestion
To enable pprof endpoints, you need to set a `pprofBindAddress` in the [manager's configuration](/docs/installation/#install-a-custom-configured-released-version).
```

File: site/content/en/docs/installation/_index.md [hunk 3]
```diff
@@ -206,3 +207,21 @@ To uninstall Kueue, run the following command:
...
+
+The easiest way to reach pprof port in kubernetes it's to use `port-forward` command:
```
Comment (alculquicondor) [inline site/content/en/docs/installation/_index.md:217]:
```suggestion
The easiest way to reach pprof port in kubernetes is to use `port-forward` command:
```

File: site/content/en/docs/installation/_index.md [hunk 4]
```diff
@@ -206,3 +207,21 @@ To uninstall Kueue, run the following command:
...
+
+```shell
+kubectl port-forward kueue-controller-manager-769f96b5dc-87sf2 -n kueue-system 8082:8082
```
Comment (alculquicondor) [inline site/content/en/docs/installation/_index.md:220]: Explain how to get the pod name

File: site/content/en/docs/installation/_index.md [hunk 5]
```diff
@@ -206,3 +207,21 @@ To uninstall Kueue, run the following command:
...
+Forwarding from 127.0.0.1:8082 -> 8082
+Forwarding from [::1]:8082 -> 8082
+```
+
+The HTTP endpoint will now be available as a local port.
+
+Please see [basic usage](https://github.com/google/pprof#basic-usage) pprof command and [examples](https://pkg.go.dev/net/http/pprof#hdr-Usage_examples) of using the exposed endpoints.
```
Comment (alculquicondor) [inline site/content/en/docs/installation/_index.md:227]:
```suggestion
To learn how to use the exposed endpoint, see [pprof basic usage](https://github.com/google/pprof#basic-usage) and [examples](https://pkg.go.dev/net/http/pprof#hdr-Usage_examples).
```

**Comment by mrbobbytables** [comment]
@moficodes it has for quite some time afaik

**Comment by moficodes** [comment]
@mrbobbytables 
The bot did not add a deploy link to this PR. :(
I thought it fell off the netlify build.

**Comment by mrbobbytables** [comment]
Not sure how, but for some reason it reverted back to only doing previews for PRs targeting the main branch. Will probably need to push an empty commit to retrigger it.

File: site/content/en/docs/installation/_index.md [hunk 6]
```diff
@@ -206,3 +207,27 @@ To uninstall Kueue, run the following command:
 ```sh
 make undeploy
 ```
+
+## Enabling pprof endpoints
```
Comment (alculquicondor) [inline site/content/en/docs/installation/_index.md:211]: Also, the installation page is probably not the best place to hold this documentation.

Can you start a new `task` https://kueue.sigs.k8s.io/docs/tasks/

Comment (alculquicondor) [reply site/content/en/docs/installation/_index.md:211]: Let's take kubernetes as inspiration https://kubernetes.io/docs/tasks/

I think it's time to divide our tasks in 2 (maybe 3?) subpages:
- Batch Admin
- Batch User

Profiling could be under "Batch Admin", but I wonder if we could add another one: "Platform Engineer/Ops"?

@cortespao any ideas?

Comment (alculquicondor) [reply site/content/en/docs/installation/_index.md:211]: I discussed this with some colleagues, and the conclusion is that there are 3 personas, so we can have a subpage for each:

- `Batch Application Developer` (previously known as Batch User).
- `Batch Ops` (previously known as Batch Administrator)
- `Batch Platform Developer` (here goes the pprof doc)

But we can leave this for a follow up.
For now, it suffices to put this in the `tasks` folder and list it under Batch Admin.

Comment (alculquicondor) [reply site/content/en/docs/installation/_index.md:211]: I opened https://github.com/kubernetes-sigs/kueue/issues/1007

File: site/content/en/docs/tasks/_index.md [hunk 7]
```diff
@@ -24,7 +24,35 @@ quotas and queues.
 - As a batch administrator, you can learn how to setup
   [Sequential Admission with Ready Pods](/docs/tasks/setup_sequential_admission).
 
-### Batch user
+### Enabling pprof endpoints
```
Comment (alculquicondor) [inline site/content/en/docs/tasks/_index.md:27]: new page please

**Review by alculquicondor — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**